### DISCUSSION TITLE
Is there a way to avoid Android dimmering screen brightness while app is active?

### DISCUSSION THREAD

**Discussion by PippoApps** [airsdk/Adobe-Runtime-Support#2283] [Q&A]
Hello folks,
I have a generative art app that Android doesn't recognize as a video player, and constantly after one minutes lowers the screen brightness.
Do you have any suggestion to prevent that?
Cheers

**Comment by ajwfrost** [comment]
I'd be wondering whether @marchbold has any existing solutions for this .. 
I assume you've tried https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/desktop/NativeApplication.html#systemIdleMode with the appropriate Android permissions?

**Reply by marchbold** [reply]
If the systemidlemode isn't working for you, you probably need to set the [`FLAG_KEEP_SCREEN_ON`](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_KEEP_SCREEN_ON) flag. If you have access to our Application ANE I can post a quick update to it to set this flag for you?

**Comment by Ender22** [comment]
I use NativeApplication.nativeApplication.systemIdleMode = SystemIdleMode.KEEP_AWAKE; to prevent the screen from turning off during videos, but can't remember if prevents dimming or not. Have you tried that?

**Reply by PippoApps** [reply]
No luck. It doesn't prevent neither dimming or turning the screen off.
My app is not a video player though, I think Android somehow can detect if a video is playing.

**Reply by PippoApps** [reply]
It seems that screen dimming can be achieved only with the flag suggested by @marchbold 
![image](https://user-images.githubusercontent.com/29799572/211211080-b7b7fee4-c6a4-4428-a893-03d91df70ce1.png)
Though, it also states that is the equivalent of:
![image](https://user-images.githubusercontent.com/29799572/211211106-80e5bbdb-d039-4086-aed8-bafeb73710ae.png)
So if there is a way to add the RelativeLayout node to the manifest it can be achieved  (althouhh being part of the UI I am afraid it is not possible).

**Comment by PippoApps** [comment]
Thank you very much for all the suggestions.
I'll have a chance to test on Friday and let you know the outcome.

**Comment by PippoApps** [comment]
No luck with systemIdleMode = SistemIdleMode.KEEP_AWAKE;
When on battery, screen dims and switches off as normally.
Android documentations suggests this:
```
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:keepScreenOn="true">
    ...
</RelativeLayout>
```

@ajwfrost is it possible to add the RelativeLayout tag in the manifest?
Thx

**Comment by MalacTheLittle** [comment]
Try adding next two lines in manifest in the combination with 
`NativeApplication.nativeApplication.systemIdleMode = SystemIdleMode.KEEP_AWAKE;`.
```
	<uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>
	<uses-permission android:name="android.permission.WAKE_LOCK"/>
```
That works fine for me.

**Reply by PippoApps** [reply]
That did the trick thank you!
The missing permissions allowed it to work.
I'll resort to Distriqt ANE just to max brightness.

**Reply by marchbold** [reply]
The wake lock permission is definitely needed for an app for remain awake. It is documented in the docs for reference:

https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/desktop/SystemIdleMode.html#KEEP_AWAKE
